idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtClassMember.kt

```diff
@@ -0,0 +1,269 @@
+), ClassMember {
+    val symbol = memberInfo.symbol
+    override fun getParentNodeDelegate(): MemberChooserObject? =
+        memberInfo.containingSymbolText?.let {
+            KtClassOrObjectSymbolChooserObject(
+                memberInfo.containingSymbolText,
+                memberInfo.containingSymbolIcon
+            )
+        }
+}
+
+private data class KtClassOrObjectSymbolChooserObject(
+    val symbolText: String?,
+    val symbolIcon: Icon?
+) :
+    MemberChooserObjectBase(symbolText, symbolIcon)
+
+internal fun createKtClassMember(
+    memberInfo: KtClassMemberInfo,
+    bodyType: BodyType,
+    preferConstructorParameter: Boolean
+): KtClassMember {
+    return KtClassMember(memberInfo, bodyType, preferConstructorParameter)
+}
+
+internal fun KtAnalysisSession.generateMember(
+    project: Project,
+    ktClassMember: KtClassMember,
+    targetClass: KtClassOrObject?,
+    copyDoc: Boolean,
+    mode: MemberGenerateMode = MemberGenerateMode.OVERRIDE
+): KtCallableDeclaration = with(ktClassMember) {
+    val bodyType = when {
+        targetClass?.hasExpectModifier() == true -> BodyType.NO_BODY
+        (symbol as? KtPossibleExtensionSymbol)?.isExtension == true && mode == MemberGenerateMode.OVERRIDE -> BodyType.FROM_TEMPLATE
+        else -> bodyType
+    }
+
+    val renderOptions = when (mode) {
+        MemberGenerateMode.OVERRIDE -> RenderOptions.overrideRenderOptions
+        MemberGenerateMode.ACTUAL -> RenderOptions.actualRenderOptions
+        MemberGenerateMode.EXPECT -> RenderOptions.expectRenderOptions
+    }
+    if (preferConstructorParameter && symbol is KtPropertySymbol) {
+        return generateConstructorParameter(project, symbol, renderOptions, mode == MemberGenerateMode.OVERRIDE)
+    }
+
+
+    val newMember: KtCallableDeclaration = when (symbol) {
+        is KtFunctionSymbol -> generateFunction(project, symbol, renderOptions, bodyType, mode == MemberGenerateMode.OVERRIDE)
+        is KtPropertySymbol -> generateProperty(project, symbol, renderOptions, bodyType, mode == MemberGenerateMode.OVERRIDE)
+        else -> error("Unknown member to override: $symbol")
+    }
+
+    when (mode) {
+        MemberGenerateMode.ACTUAL -> newMember.addModifier(KtTokens.ACTUAL_KEYWORD)
+        MemberGenerateMode.EXPECT -> if (targetClass == null) {
+            newMember.addModifier(KtTokens.EXPECT_KEYWORD)
+        }
+        MemberGenerateMode.OVERRIDE -> {
+            // TODO: add `actual` keyword to the generated member if the target class has `actual` and the generated member corresponds to
+            //  an `expect` member.
+        }
+    }
+
+    if (copyDoc) {
+        val kDoc = when (val originalOverriddenPsi = symbol.originalOverriddenSymbol?.psi) {
+            is KtDeclaration ->
+                findDocComment(originalOverriddenPsi)
+            is PsiDocCommentOwner -> {
+                val kDocText = originalOverriddenPsi.docComment?.let { IdeaDocCommentConverter.convertDocComment(it) }
```

Comment:
Please, move IdeaDocCommentConverter to frontend-independent module

Reply:
Thanks! I missed that!